### PR TITLE
[flutter_tools] For l10n with deferred loading, use loadLibrary for non-web too

### LINKED ISSUE #59509: [l10n] skipped test due to changes Dart VM support for deferred loading
The Dart VM/runtime may have support for deferred loading libraries in the future. This will require deferred loaded code to correctly use the `loadLibrary()` functionality before references imported members.

As a result of this change, the second test in the l10n generation integration test will be skipped: https://github.com/flutter/flutter/pull/59504

We should determine the code update needed to unskip, or whether the test is valuable.

@HansMuller @shihaohong

### PR DESCRIPTION
**Pull request by perclasson**
After Dart VM change we are now required to use loadLibrary on an import whenever it is imported as deferred.

See:
https://dart-review.googlesource.com/c/sdk/+/149613

Tried it out in the Gallery:
https://github.com/flutter/gallery/pull/183

## Description

This PR makes it so that when you are using the gen l10n with deferred loading, the deferred imports will have loadLibrary called on them regardless if its web or not.

The integration test update is needed as deferred localizations are loaded asynchronously and not with SynchronousFuture anymore, and the [Localizations](https://api.flutter.dev/flutter/widgets/Localizations-class.html) widget is rebuilt when each language is loaded. In an app you usually only have localizations for one language used at a time so this would not be noticed, but here all supported locales are run in the same widget.

For more detail see: 
https://github.com/flutter/flutter/blob/4d7525f05c05a6df0b29396bc9eb78c3bf1e9f89/packages/flutter/lib/src/widgets/localizations.dart#L517

By using the LocaleBuilder for the all of the printing widgets inside of `gen_l10n_project.dart`, they all have to wait for the localizations to load before printing any result and the test passes.

## Related Issues

Closes https://github.com/flutter/flutter/issues/59509
## Tests

I added the following tests:

- imports are deferred and loaded when useDeferredImports are set

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I signed the [CLA].
- [X] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] All existing and new tests are passing.
- [X] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [X] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [X] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: https://flutter.dev/go/breaking-changes-template *Replace this with a link to a pull request that adds the migration guide to https://flutter.dev/docs/release/breaking-changes*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter_tools/test/integration.shard/test_data/gen_l10n_project.dart [hunk 1]
```diff
@@ -94,7 +94,7 @@ class Home extends StatelessWidget {
     final List<String> results = [];
     return Row(
       children: <Widget>[
-        ResultBuilder(
+        LocaleBuilder(
```
Comment (rami-a) [inline packages/flutter_tools/test/integration.shard/test_data/gen_l10n_project.dart:97]: Are the changes in this file related to deferred loading?

**Review by rami-a — APPROVED** [review]

File: packages/flutter_tools/test/integration.shard/test_data/gen_l10n_project.dart (hunk 1 above)
Comment (perclasson) [reply packages/flutter_tools/test/integration.shard/test_data/gen_l10n_project.dart:97]: Yes. It was quite hard to pin this down. If you look at the description of the PR you can see the full details.

The TLDR is that this change is needed as the `Localizations` widget rebuilds itself after localizations have loaded. This change ensure that all of the printing widgets in the test will wait for all localizations to load first.

**Review by jonahwilliams — APPROVED** [review]
LGTM

Thank you for fixing this!

**Comment by jonahwilliams** [comment]
Landing to unblock dart

**Pull request merged**